packages/docusaurus-theme-classic/src/theme/BlogPostItem/Header/Author/index.tsx

```diff
@@ -5,11 +5,70 @@
  * LICENSE file in the root directory of this source tree.
  */
 
+import type {SVGProps} from 'react';
 import React from 'react';
 import clsx from 'clsx';
 import Link, {type Props as LinkProps} from '@docusaurus/Link';
 
 import type {Props} from '@theme/BlogPostItem/Header/Author';
+import styles from './styles.module.css';
+
+type SocialProps = SVGProps<SVGSVGElement> & {
+  user: string;
+};
+
+function getUrl(user: string, baseUrl: string) {
+  return user.startsWith('https') ? user : `${baseUrl}${user}`;
+}
+
+function Twitter(props: SocialProps) {
```

Comment:
export in its own file/folder

Reply:
Yes, a good place could be theme-classic `theme/Icon/Social/Twitter`, since we already have an `Icon` folder

Note: only the SVG should be added there as a reusable icon, not the link.